python/ray/data/_internal/dataset_logger.py

```diff
@@ -71,5 +80,7 @@ def get_logger(self, log_to_stdout: bool = True):
         also removes the need for this getter method:
         `logger.info(msg="Hello world", stacklevel=2)`
         """
+        if not self._logger:
+            self._logger = self._initialize_logger()
```

Comment:
The below `self._logger.propagate = log_to_stdout` will fail with an `AttributeError` if `self._logger` is `None` (i.e. if `DatasetLogger().get_logger()` is called before `ray.init()`), I think we should explicitly raise an error in this case.
```suggestion
            self._logger = self._initialize_logger()
            if self._logger is None:
                raise ValueError("DatasetLogger.get_logger() must be called after ray.init().")
```

This is assuming that we allow `_initialize_logger()` to return `None` in that case, see above comment for another option.

Reply:
Went with the other option, where we raise the `ValueError` within `_initialize_logger()` if `ray.init()` had not yet been called (i.e. `global_node` is None).